lldb/packages/Python/lldbsuite/test/lldbplatformutil.py

```diff
@@ -92,11 +96,22 @@ def match_android_device(device_arch, valid_archs=None, valid_api_levels=None):
 
 
 def finalize_build_dictionary(dictionary):
+    if dictionary is None:
+        dictionary = {}
     if target_is_android():
-        if dictionary is None:
-            dictionary = {}
         dictionary["OS"] = "Android"
         dictionary["PIE"] = 1
+    elif platformIsDarwin():
+        dictionary["OS"] = "Darwin"
+    else:
+        # Provide uname-like platform name
+        platform_name_to_uname = { "linux": "Linux",
+                "netbsd": "NetBSD",
+                "freebsd": "FreeBSD",
+                "windows": "Windows_NT",
+                }
+        dictionary["OS"] = platform_name_to_uname[getPlatform()]
+        dictionary["HOST_OS"] = platform_name_to_uname[getHostPlatform()]
```

Comment:
Why is this under the `else:` branch? I would expect we need to set `HOST_OS` even if we're targetting android or darwin...